mbed_flasher/erase.py

```diff
@@ -67,90 +63,44 @@ def __get_flashers():
         from mbed_flasher.flashers import AvailableFlashers
         return AvailableFlashers
 
-    def reset_board(self, serial_port):
+    @staticmethod
+    def _can_be_erased(target):
         """
-        :param serial_port: board serial port
-        :return: return exit code based on if successfully reset a board
+        Check if target can be erased.
+        :param target: target board to be checked
+        :return: None if can be erased, raises otherwise
         """
-        from mbed_flasher.flashers.enhancedserial import EnhancedSerial
-        from serial.serialutil import SerialException
-
         try:
-            port = EnhancedSerial(serial_port)
-        except SerialException as err:
-            self.logger.error(err)
-            # SerialException.message is type "string", it has 'find'
-            #  pylint: disable=no-member
-            if err.message.find('could not open port') != -1:
-                self.logger.error(
-                    'Reset could not be given. Close your Serial connection to device.')
-
-            raise EraseError(message="reset could not be sent",
-                             return_code=EXIT_CODE_SERIAL_PORT_OPEN_FAILED)
-
-        port.baudrate = 115200
-        port.timeout = 1
-        port.xonxoff = False
-        port.rtscts = False
-        port.flushInput()
-        port.flushOutput()
-        if port:
-            self.logger.info("sendBreak to device to reboot")
-            result = port.safe_send_break()
-            if result:
-                self.logger.info("reset completed")
-            else:
-                raise EraseError(message="reset failed",
-                                 return_code=EXIT_CODE_SERIAL_RESET_FAILED)
-
-        port.close()
-        return EXIT_CODE_SUCCESS
-
-    def wait_to_disappear(self, mount_point):
-        """
-        :param mount_point: mount_point to watch disappear
-        """
-        start_time = time()
-        while True:
-            sleep(0.1)
-            if platform.system() == 'Windows':
-                proc = Popen(["dir", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE, shell=True)
-            else:
-                proc = Popen(["ls", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE)
-
-            err = proc.stderr.read()
-            proc.communicate()
-
-            if err:
-                self.logger.debug("Remount due to erase")
-                break
-
-            if time() - start_time > ERASE_REMOUNT_TIMEOUT:
-                self.logger.debug("Didn't notice device to disappear for remount")
-                break
+            with open(join(target["mount_point"], 'DETAILS.TXT'), 'rb') as new_file:
+                details_txt = new_file.readlines()
+        except IOError:
```

Comment:
For python 3, here will raise OSError for open, [doc](https://docs.python.org/3/library/functions.html#open)